src/utils/gltf/gltfParser.ts

```diff
@@ -0,0 +1,518 @@
+import { Entity } from "../../entity";
+import { Quat } from "../../math/Quat";
+import { Vec3 } from "../../math/Vec3";
+import { Object3d } from "../../Object3d";
+import { Glb } from "./glbParser";
+import {
+    Accessor,
+    AccessorComponentType,
+    AccessorDataType,
+    GltfData,
+    TextureImage,
+    Material,
+    Mesh,
+    Primitive,
+    PrimitiveMode,
+    Texture,
+    MimeType,
+    GltfNode,
+    GltfMesh,
+    GltfPrimitive
+} from "./types";
+
+export class Gltf {
+    private static dracoDecoderModule: any = null;
+    public static connectDracoDecoderModule(decoder: any): void {
+        this.dracoDecoderModule = decoder;
+    }
+    public static async loadGlb(url: string) {
+        const data = await Glb.load(url);
+        if (this.dracoDecoderModule !== null) {
+            await this.dracoDecoderModule.ready;
+        }
+        return new Gltf(data);
+    }
+
+    private materials: Material[] = [];
+    private images: TextureImage[] = [];
+    public meshes: Mesh[] = [];
+
+    constructor(private gltf: GltfData) {
+        if (
+            gltf.gltf.extensionsRequired?.includes("KHR_draco_mesh_compression") &&
+            Gltf.dracoDecoderModule === null
+        ) {
+            throw new Error("Unable to import GLTF. Draco decoder module is not connected");
+        }
+        this.initImages();
+        this.initMaterials();
+        this.initMeshes();
+    }
+
+    public getObjects3d(): Object3d[] {
+        return this.meshes
+            .map((mesh) => mesh.primitives.map((primitive) => Gltf.toObject3d(primitive)))
```

Comment:
Just in case (no need to change right now) if we ever run into a model with a billion meshes and primitives, I'd consider using for loops instead of map, for performance reasons.

```
    for (let i = 0, leni = this.meshes.length; i < leni; i++) {
        let mi = this.meshes[i];
        for (let j = 0, lenj=mi.primitives; j < lenj; j++) {
            let pi = mi.primitives[j];
            res.push(Gltf.toObject3d(primitive));
        }
    }
```

P.S.
Some people wouldn't agree with me:)